### PR TITLE
Use a more consistent naming convention for bid status presenters

### PR DESCRIPTION
**Pull request by jessieay**
* First namespace is auction status
* Second / third  namespace, if any, is the user type
* Only use role namespace if there are multiple presenters for that user type

### CODE REVIEW

File: app/models/bid_status_presenter_factory.rb [hunk 1]
```diff
@@ -22,67 +22,67 @@ def create
...
     else
-      BidStatusPresenter::OverUserIsWinner
+      BidStatusPresenter::Over::Vendor::Winner::WorkNotStarted
     end
   end
 
   def over_message
     if user_is_bidder?
-      BidStatusPresenter::OverUserIsBidder
+      BidStatusPresenter::Over::Vendor::Bidder
```
Comment (harrisj) [inline app/models/bid_status_presenter_factory.rb]: Holy namespaces, Batman!

**Comment by harrisj** [comment]
At first, I found myself a bit 😕 about five layers of namespacing for some classes, but it really is equivalent to having a long and unwieldy class name, so I support this. But I want to give others a chance to weigh in

**Comment by jessieay** [comment]
Yes it is a bit annoying when opening up a file, but the nice thing about namespacing like this is that it forces consistency because you don't want to create all kinds of dirs for different types of presenters that are for the same user type etc. Imagining that the next version of this will include some type of interpolation based on the auction status and person viewing the auction

**Comment by harrisj** [comment]
Just make sure it's automagical interpolation and loading to keep it Railsy

**Comment by jessieay** [comment]
ping ping @adelevie , jacob is requesting your 👀 on this one

**Comment by adelevie** [comment]
Yeah I think this is fantastic. If stuff is unwieldy, better to name it and make it clear than to hide it in longer module/class names.

**Pull request merged**